nhais-adaptor/edifact/incoming/parser/tests/test_deserialiser.py

```diff
@@ -4,124 +4,27 @@
-            ])
-
-            message_registration_dict = deserialiser.extract_relevant_lines(original_dict, 9, "S02")
-
-            compare(message_registration_dict, expected)
-
-    def test_convert_to_dict(self):
-        with self.subTest("Without duplicate keys"):
-            expected = EdifactDict([
-                ("UNB", "UNOA:2+SO01+ROO5+190429:1756+00016288++FHSREG+++FHSA EDI TRANSFERS"),
-                ("UNH", "00024986+FHSREG:0:1:FH:FHS001"),
-                ("BGM", "++507")
-            ])
-
-            input_lines = [
-                "UNB+UNOA:2+SO01+ROO5+190429:1756+00016288++FHSREG+++FHSA EDI TRANSFERS",
-                "UNH+00024986+FHSREG:0:1:FH:FHS001",
-                "BGM+++507"
-            ]
-            converted_dict = deserialiser.convert_to_dict(input_lines)
-
-            compare(converted_dict, expected)
-
-        with self.subTest("With duplicate keys"):
-            expected = EdifactDict([
-                ("NAD", "FHS+SO:954"),
-                ("DTM", "137:201904291755:203"),
-                ("NAD", "GP+1231231,PLP348:900")
-            ])
-
-            input_lines = [
-                "NAD+FHS+SO:954",
-                "DTM+137:201904291755:203",
-                "NAD+GP+1231231,PLP348:900",
-            ]
-            converted_dict = deserialiser.convert_to_dict(input_lines)
-
-            compare(converted_dict, expected)
-
     def test_convert(self):
         with self.subTest("When the edifact incoming message does not have a patient section (S02)"):
-            expected = Interchange(InterchangeHeader("SO01", "ROO5", "190429:1756"), Messages([
-                MessageSegment(MessageSegmentBeginningDetails("F4"), MessageSegmentRegistrationDetails("211102"))
-            ]))
+            expected = Interchange(InterchangeHeader("SO01", "ROO5", "190429:1756"),
+                                   Messages([
+                                       MessageSegment(MessageSegmentBeginningDetails("F4"),
+                                                      Transactions([
+                                                          Transaction(
+                                                              TransactionRegistrationDetails("211102")
+                                                          )
+                                                      ]))
+                                   ]))
 
-            input_lines = [
-                "UNB+UNOA:2+SO01+ROO5+190429:1756+00016288++FHSREG+++FHSA EDI TRANSFERS",
-                "UNH+00024986+FHSREG:0:1:FH:FHS001",
-                "BGM+++507",
-                "NAD+FHS+SO:954",
-                "DTM+137:201904291755:203",
-                "RFF+950:F4",
-                "S01+1",
-                "RFF+TN:211102",
-                "NAD+GP+1231231,PLP348:900",
-                "UNT+9+00024986",
-                "UNZ+1+00016288",
-            ]
+            input_lines = LineBuilder().start_interchange().start_message("F4")\

```

Comment:
I generally like my builder calls in the form:
```python
input_lines = LineBuilder()\
    .start_interchange()\
    .start_message("F4")\
    .add_transaction("211102")\
    .end_message()\
    .end_interchange()\
    .build()
```
So that all of the calls being made are obvious. Do you think that's clearer in this instance?

Reply:
I've put it on separates lines as suggested and I think it does read better 👍